### PR TITLE
ENG-7825 - Add new startAppFlow Command

### PR DESCRIPTION
**Pull request by kevinsites**
Add new command for multi use case in one app
Will work with existing start/startSession commands
Adds `linkedSiteId` to packet header for backend use. This will be cleared on stop/stopSession but not before.

### CODE REVIEW

**Comment by kevinsites** [comment]
Sonar error is on test duplication but should be ignored

File: NeuroID/NeuroIDClass/NeuroID.swift [hunk 1]
```diff
@@ -175,8 +177,9 @@ public enum NeuroID {
             return false
         }
 
-        NeuroID.groupAndPOST()
+        NeuroID.groupAndPOST(forceSend: true)
         NeuroID._isSDKStarted = false
+        NeuroID.linkedSiteID = nil
```
Comment (jeff-niemann) [inline NeuroID/NeuroIDClass/NeuroID.swift:182]: Any chance where customer could call `startAppFlow()` and then `stop()` and then `start()`? Is the intention that if they are configured for multi-flows they will always call `startAppFlow()` and never `start()`

File: NeuroID/NeuroIDClass/Extensions/NIDSession.swift [hunk 2]
```diff
@@ -206,4 +208,37 @@ public extension NeuroID {
 
         return true
     }
+
+    static func startAppFlow(siteID: String, userID: String? = nil) -> SessionStartResult {
+        if NeuroID.clientKey == nil || NeuroID.clientKey == "" {
```
Comment (jeff-niemann) [inline NeuroID/NeuroIDClass/Extensions/NIDSession.swift:213]: This check could be a function to reduce code duplication

File: NeuroID/NeuroIDClass/Extensions/NIDSession.swift [hunk 3]
```diff
@@ -206,4 +208,37 @@ public extension NeuroID {
...
+            NIDLog.e("Missing Client Key - please call configure prior to calling start")
+            return SessionStartResult(false, "")
+        }
+
+        // immediately flush events before anything else
+        groupAndPOST(forceSend: true)
```
Comment (jeff-niemann) [inline NeuroID/NeuroIDClass/Extensions/NIDSession.swift:219]: Is this still okay if we are in a `stopped` state? 
I agree queue should be flushed before anything else but just wanted to check if it was safe to do if `stopped`

File: NeuroID/NeuroIDClass/Extensions/NIDSession.swift [hunk 4]
```diff
@@ -206,4 +208,37 @@ public extension NeuroID {
...
+                startStatus = SessionStartResult(started, "")
+            }
+        } else {
+            startStatus = SessionStartResult(true, NeuroID.getUserID())
+        }
+
+        // add linkedSite var
+        NeuroID.linkedSiteID = siteID
+
+        let setLinkedSiteIDEvent = NIDEvent(type: NIDEventName.closeSession)
```
Comment (jeff-niemann) [inline NeuroID/NeuroIDClass/Extensions/NIDSession.swift:237]: Why is the event type `NIDEventName.closeSession`?

File: NeuroID/NeuroIDClass/NeuroID.swift (hunk 1 above)
Comment (kevinsites) [reply NeuroID/NeuroIDClass/NeuroID.swift:182]: There is a chance yes. But if they are configured for multi-flows they should call start/startSession/startAppFlow 
-> startAppFlow
-> startAppFlow
-> stop/stopSession

File: NeuroID/NeuroIDClass/Extensions/NIDSession.swift (hunk 3 above)
Comment (kevinsites) [reply NeuroID/NeuroIDClass/Extensions/NIDSession.swift:219]: Yes I think even if we are in a "stopped" state meaning we are collecting anymore we should still flush the events. This also fixes a very minor race condition when stop/stopSession is called where we send the packet of events and then capture a CLOSE_SESSION event but never send it

File: NeuroID/NeuroIDClass/Extensions/NIDSession.swift (hunk 4 above)
Comment (kevinsites) [reply NeuroID/NeuroIDClass/Extensions/NIDSession.swift:237]: good catch, fixing

File: NeuroID/NeuroIDClass/Extensions/NIDSession.swift [hunk 5]
```diff
@@ -206,4 +208,37 @@ public extension NeuroID {
...
+        // immediately flush events before anything else
+        groupAndPOST(forceSend: true)
+
+        // If not started then start
+        var startStatus: SessionStartResult
+        if !NeuroID._isSDKStarted {
+            if userID != nil {
+                startStatus = NeuroID.startSession(userID)
+            } else {
+                let started = NeuroID.start()
```
Comment (priya-xavier) [inline NeuroID/NeuroIDClass/Extensions/NIDSession.swift:227]: what's the reasoning behind using start() here instead os startSession()? Both would create a new session id right?

Comment (kevinsites) [reply NeuroID/NeuroIDClass/Extensions/NIDSession.swift:140]: No, start will not create a sessionID which for some use cases is appropriate (I.e. MGI and Aspiration)

**Review by priya-xavier — APPROVED** [review]

**Review by jeff-niemann — APPROVED** [review]

**Pull request merged**